mm2src/mm2_main/src/lp_swap/swap_watcher.rs

```diff
@@ -193,6 +192,7 @@ impl State for ValidateTakerFee {
     }
 }
 
+// TODO: Validate also maker payment
```

Comment:
Yes it does for the ETH transactions, I moved that logic inside the `watcher_validate_taker_payment` method to get rid of the `taker_coin.is_eth`. 

https://github.com/KomodoPlatform/atomicDEX-API/blob/fad0116935cde05648d9ccfca025b37c368eef50/mm2src/coins/eth.rs#L1527-L1531

https://github.com/KomodoPlatform/atomicDEX-API/blob/fad0116935cde05648d9ccfca025b37c368eef50/mm2src/coins/eth.rs#L1597-L1604

For UTXO it doesn't because the transaction will contain the trade amount + watcher reward, and the watcher doesn't know about the trade amount. But some validation can still be implemented by deriving the trade amount from the taker fee.